PRINCIPLES.md

```diff
@@ -0,0 +1,254 @@
+The Desired State of the new version, Who created it, when it was created and ideally, also _why_ it was created. - What Why, who when. 
+
+- Even though we say git, it's actually only true when configured in a very particular way.
+
+#### State stores as coordination points/ collab via common data interface
+- Collab point. 
+- Business rules and security rules about the acceptable state of the system shoudl be checked against any new proposed version of the desired configuration before such changes are accepted as a new version. (Gates before version)
+- where human process is integrated
+- the place where collaboration can be synchronised.
+-->
+
+### 3. Continuous state reconciliation
+
+<div style="background: #E3F2FD; border-left: 3px solid #0D47A1; padding:10px; margin-bottom:5px; font-style:italic">
+Software agents continuously, and automatically, compare a system's <em>Actual State</em> to its <em>Desired State</em>.
+If the actual and desired states differ, automated actions are immediately attempted to reconcile them.
+These differences could be due to the actual state drifting from the desired state, or the desired state changing intentionally.
+</div>
+
+<!--
+- If the software agents fail to bring the system's state in line with its desired state, a human operator is notified.
+- Software agents continuously check that the running system under management matches the desired state configuration, and if it does not, immediately either take remedial action to bring the system back in line with stated expectations or, if this cannot be done, alert a human operator that the system is no longer meeting expectations
+- Automated delivery: Delivery of the declarative descriptions, from the repository to runtime environment, is fully automated.
+- Software Agents: Reconcilers maintain system state and apply the resources described in the declarative configuration.
+- Closed loop: Actions are performed on divergence between the version controlled declarative configuration and the actual state of the target system.
+- This automation should serve its human users.
+  To this point, reconciliation automation for GitOps should include a way for humans to temporarily take back the reins as needed.
+- Implies observability. State of the running system must be observable. Ideally in same format as the configuration so we can compare like to like. 
+- >I agree. I think of this as an observability requirement and tend to say that "the current state of a running system must be observable" to mean the same thing. This is, in my opinion, required for state reconciliation. I didn't add a note there explicitly though, as that section was unfinished.
+-->
+
+
+### 4. Operations through declaration
+
+<div style="background: #E3F2FD; border-left: 3px solid #0D47A1; padding:10px; margin-bottom:5px; font-style:italic">
+When wishing to operate on a software system, a human or software agent will not interact with the running system and modify it directly.
```

Comment:
Isn't this incorrect? A software agent WILL interact with the system to apply the declaration that the human committed to the state store? (think an operator in a cluster)

Reply:
```suggestion
When wishing to operate on a software system, a human or computer program will not interact with the running system and modify it directly. Only the reconciliation-agent will operate on the software system.
```

Reply:
```suggestion
The mechanism in which change is applied to a system by either a human operator or another system is through a declarative change committed to a state store.  Reconciliation agents then apply these changes to the systems they control. 
```